admin/app/components/solidus_admin/ui/table/ransack_filter/component.js

```diff
@@ -5,13 +5,34 @@ export default class extends Controller {
   static targets = ['details', 'summary', 'option']
 
   connect() {
+    this.setCheckboxesFromUrlParams()
     useClickOutside(this)
   }
 
   clickOutside(event) {
     this.detailsTarget.removeAttribute("open")
   }
 
+  setCheckboxesFromUrlParams() {
+    let checkboxSelected = false
+    const urlParams = new URLSearchParams(window.location.search)
+
+    for (const [key, value] of urlParams) {
+      if (key.endsWith("[v][]")) {
+        const checkbox = this.element.querySelector(`input[name="${key}"][value="${value}"]`)
+        if (checkbox) {
+          checkboxSelected = checkbox.checked = true
+        }
+      }
+    }
+
+    if (checkboxSelected) this.dispatchShowSearchEvent()
+  }
```

Comment:
The main reason was simplicity 😄  However, I now check the params in the component and render the checkboxes as already selected.
I had to introduce a debounce because I noticed that with quick page refreshes, the controller sometimes doesn't wait for the complete rendering of the checkboxes and doesn't trigger `showSearch` (I couldn't find a better solution at the moment).